pgjdbc/src/test/java/org/postgresql/test/jdbc2/DatabaseMetaDataTest.java

```diff
@@ -1961,6 +1961,44 @@ void functionColumns(BinaryMode binaryMode) throws SQLException {
     rs.close();
   }
 
+  @MethodSource("data")
+  @ParameterizedTest(name = "binary = {0}")
+  void charOctetLengthIsNullForNonCharTypes(BinaryMode binaryMode) throws SQLException {
+    Assumptions.assumeTrue(TestUtil.haveMinimumServerVersion(con, ServerVersion.v9_2));
+    TestUtil.createTable(con, "char_octet_test", "intarray integer[], mytext varchar(100), mychar char(10), mybinary bytea");
+    DatabaseMetaData dbmd = con.getMetaData();
+
+    // integer[] -> null
+    ResultSet rs = dbmd.getColumns(null, "public", "char_octet_test", "intarray");
+    assertTrue(rs.next());
+    assertEquals("intarray", rs.getString("COLUMN_NAME"));
+    assertNull(rs.getString("CHAR_OCTET_LENGTH"), "Expected Null for integer[]");
+    rs.close();
+
+    // varchar(100) -> 100
+    rs = dbmd.getColumns(null, "public", "char_octet_test", "mytext");
+    assertTrue(rs.next());
+    assertEquals("mytext", rs.getString("COLUMN_NAME"));
+    assertEquals("100", rs.getString("CHAR_OCTET_LENGTH"), "Expected 100 for varchar(100)");
```

Comment:
Same thing, `expected 100` will appear in the message automatically.

```suggestion
    assertEquals("100", rs.getString("CHAR_OCTET_LENGTH"), "CHAR_OCTET_LENGTH for varchar(100)");
```